testsuite/testsuite-microprofile-jwt/src/test/java/org/wildfly/swarm/microprofile/jwtauth/RolesAllowedEmptyPropsTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *   Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package org.wildfly.swarm.microprofile.jwtauth;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.UrlAsset;
+import org.jboss.shrinkwrap.impl.base.io.IOUtil;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.jaxrs.JAXRSArchive;
+
+@RunWith(Arquillian.class)
+public class RolesAllowedEmptyPropsTest {
+
+    @Deployment
+    public static Archive<?> createDeployment() throws Exception {
+        JAXRSArchive deployment = ShrinkWrap.create(JAXRSArchive.class);
+        deployment.addResource(RolesEndpointClassLevel.class);
+        deployment.addResource(TestApplication.class);
+        deployment.addAllDependencies();
+        deployment.addAsResource("emptyRoles.properties");
+        deployment.addAsResource("project-defaults.yml");
+        URL url = RolesAllowedEmptyPropsTest.class.getResource("/publicKey.pem");
+        deployment.addAsManifestResource(new UrlAsset(url), "/MP-JWT-SIGNER");
+        return deployment;
+    }
+
+    @RunAsClient
+    @Test
+    public void testRolesAllowed() throws Exception {
+        String uri = "http://localhost:8080/mpjwt/rolesClass";
+        WebTarget target = ClientBuilder.newClient().target(uri).queryParam("input", "hello");
```

Comment:
Yea, this is why I hate JAX-RS Client: it doesn't implement `AutoCloseable`, so noone bothers to properly close it. Especially in a test.

Reply:
What exactly is the problem here from the test's point of view ? Fluent HC does not really scale. Just because it auto closes does not make a preferred candidate IMHO. It will break as soon as one needs to introduce a filter or deal with more than a String. JAX-RS 2.0 client API is rich. I agree tat it is not perfect :-), I did have a lot of unhappiness about some parts of it some time ago but it is very capable nonetheless. FYI, Response InputStream is expected to be auto-closed by the spec for the typed reads

Reply:
May be RestEasy specific Client implementation allows explicitly closing or even Auto closing, I don't know. JAX-RS Response itself has close() but I'm not sure how it is implemented in RestEasy - at te very least InputStream must be closed. And this InputStream must be closed if there was something like ".get(String.class)" - though in CXF it is only auto-closed if it was pre-configured to do so - otherwise there were side-effects with lazy reading DOM/XPath and similar types